executor/runtime/docker/docker.go

```diff
@@ -1781,6 +1783,32 @@ func imageNameAndDigest(image, containerName string) (name, digest string, err e
 	return name, digest, nil
 }
 
+func (r *DockerRuntime) generatePlatformSidecarSpectatordMetrics() []string {
+	l := log.WithField("taskID", r.c.TaskID())
+	var metricLines []string
+	for k, v := range r.c.Pod().Annotations {
+		releaseSuffix := fmt.Sprintf(".%s/%s", pod.AnnotationKeySuffixSidecars, pod.AnnotationKeySuffixSidecarsRelease)
+		if !strings.HasSuffix(k, releaseSuffix) {
+			continue
+		}
+		sidecar := strings.TrimSuffix(k, releaseSuffix)
+		release := strings.Split(v, "/") // Expected format: $channel/$channelDefID
+		if len(release) != 2 || release[0] == "" || release[1] == "" {
+			l.Warnf("Unexpected release value for platform sidecar %q: %q", sidecar, v)
+			continue
+		}
```

Comment:
I'm wondering if this annotation parsing code should live in `titus-kube-common`.

Reply:
I'm not aware of other Go code that needs to parse the release annotation. I think it's okay to leave the parsing logic here for now. We can definitely move it to `titus-kube-common` if code sharing is beneficial.

Reply:
I think in general we've tried to centralize our annotation handling code there, to abstract usage of annotations instead of spreading parsing/formatting logic over many repos. @solarkennedy can jump in and correct me if I'm wrong, though.

Reply:
Yea I guess that would be a nice to have. I'm ok either way. Certainly as soon as there is a second place that needs this parsing, it should be shared.
This is certainly non-trivial parsing of the constants.